fiftyone/utils/eval/detection.py

```diff
@@ -182,14 +192,50 @@ def evaluate_image(
                 :class:`fiftyone.core.labels.Detections` instances
             gt_field: the name of the field containing the ground truth
                 :class:`fiftyone.core.labels.Detections` instances
-            eval_key (None): an evaluation key for this evaluation
+            eval_key (None): the evaluation key for this evaluation
 
         Returns:
             a list of matched ``(gt_label, pred_label, iou, pred_confidence)``
             tuples
         """
         raise NotImplementedError("subclass must implement evaluate_image()")
 
+    def generate_results(
```

Comment:
I renamed this to `generate_results()` to clarify what it does. Otherwise it can be confusing to see that `evaluate_image` is called only to call `evaluate_samples` later.

Also, since the main `evaluate_detections()` method calls it regardless of the evaluation method being run, you need to define it on the base `DetectionEvaluation` class (even though, yes, right now only COCO method exists)

Reply:
Good call!